Extensions/one_click_postage.js

```diff
@@ -1517,25 +1426,62 @@ XKit.extensions.one_click_postage = new Object({
 		XKit.storage.set("one_click_postage", "already_reblogged", JSON.stringify(XKit.extensions.one_click_postage.already_reblogged));
 	},
 
-	show_error: function(code, state) {
-		var m_word = "reblog";
-		if (state === 1) { m_word = "draft"; }
-		if (state === 2) { m_word = "queue"; }
+	show_error: function(error, state) {
+		const verb = ["reblog", "draft", "queue"][state];
+		let message = "";
+
+		if (error.status) {
+			// HTTP error: response object
+			let response = error;
+
+			message = `The server returned HTTP ${response.status}.<br>`;
+			switch (response.status) {
+				case 200:
+					message += "However, Tumblr has indicated that something is wrong.";
+					break;
+				case 401:
+					message += "This usually means your browser is not sending Referer headers.";
+					break;
+				case 403:
+					message += `This usually means you've been blocked by the owner of the post.`;
+					break;
+				case 404:
+					message += `The post you are trying to ${verb} has been deleted.`;
+					break;
+				default:
+					message += "We're not sure what this means.<br>Please consider reporting this to us!";
+			}
 
-		var m_causes = "<ul class=\"xkit-one-click-postage-error-list\">" +
-					"<li><b>You have used your daily post limit. (<a href=\"#\">more information</a>)</b><br/>The limit is 250 per day, set by Tumblr. Try again in 24 hours.<br/>This affects the queue too, and can't be circumvented.</li>" +
-					"<li><b>You've filled up your queue.</b><br/>You can not queue more than 300 posts.</li>" +
-					"<li><b>The post is deleted.</b><br/>The post you are trying to " + m_word + " is deleted by the user.</li>" +
-					"<li><b>Your browser settings are denying XKit cookies.</b><br/>If you have disabled \"Third Party Cookies\", One-Click Postage can not work properly. Please enable them and try again.</li>" +
-					"<li><b>There was a server error / change.</b><br/>Wait for a while and retry the " + m_word + " request.<br/>Check the XKit blog for updates.</li>" +
-				"</ul>";
+			message += "<br><br>";
 
-		if (state === 2 && XKit.interface.user().queue >= 299) {
-			m_causes = "<ul class=\"xkit-one-click-postage-error-list\">" +
-					"<li><b>You've filled your queue.</b><br/>You can not queue more than 300 posts.</li>" +
-				"</ul>";
+			try {
+				let data = response.json();
+				if (data.errors !== undefined) {
+					message += `Tumblr provided this error message: <p style="overflow-x:scroll">${JSON.stringify(data.errors)}</p>`;
+				} else if (data.error !== undefined) {
+					message += `<p>${data.error}</p>`;
+				}
+			} catch (e) {
+				message += "Tumblr returned a generic HTTP error page. Please ensure you are logged in.";
```

Comment:
In spirit of making these more actionable, maybe "Refresh the dashboard and try again" makes more sense?